Model/User.php

```diff
@@ -190,9 +176,7 @@ public function unserialize($serialized)
             $this->salt,
             $this->usernameCanonical,
             $this->username,
-            $this->expired,
             $this->locked,
-            $this->credentialsExpired,
```

Comment:
The unserialize method already has an `array_fill()` from a long time ago (sha: e1f5b11ba77bb24f2d47054e7ea41e0ba8c1e2a2). And I think this *wasn't* adjusted in #1936, so technically I think we broke serialization in that PR. There's no release yet containing that change... but since everyone is using `dev-master`... probably a lot of people are already on that version. If I'm right, we have no choice to break that BC in favor of trying BC for `dev-master`** in order to keep BC for the actual last, stable release.

The last stable release had 9 properties in unserialize (https://github.com/FriendsOfSymfony/FOSUserBundle/blob/d66890ad3489e18be153502c5ccc3f2bf5cce442/Model/User.php). So, what if we:

A) Remove these fields from serialize, exactly as you see here
B) In unserialize, count the items in `$data`. If there are 9 (the old number), then remove keys 4 and 6 (i.e. remove `expired` and `credentialsExpired`), `array_fill()` for the 4 properties from #1936 and then set the properties. If there are 11 [9 - 2 properties removed + 4 properties added in #1936] the new number), skip this operation.

\*\* Technically, we could probably keep BC with `dev-master` also, by looking for 13 items and removing indexes 4 & 6, but not using `array_fill`.

Also, is it legal to use an associative array instead of an index array? We could change serialize to serialize an associative array instead, and still do the above magic to help "migrate" old, indexed data.

Can anyone find an issue with this plan?